lib/wallet/http.js

```diff
@@ -200,6 +200,26 @@ class HTTP extends Server {
       await this.wdb.rescan(height);
     });
 
+    // Deep Rescan
+    this.post('/deeprescan', async (req, res) => {
+      if (!req.admin) {
+        res.json(403);
+        return;
+      }
+
+      const valid = Validator.fromRequest(req);
+      const disclaimer = valid.bool('I_HAVE_BACKED_UP_MY_WALLET', false);
+
+      enforce(
+        disclaimer,
+        'Deep rescan requires I_HAVE_BACKED_UP_MY_WALLET=true'
+      );
+
+      res.json(200, { success: true });
```

Comment:
Why is the response being set before the `deepRescan()` call? Is it to avoid caller timeout since it's a long-running operation?

Reply:
Exactly. You see the same logic in the regular `/rescan` endpoint.

Reply:
:+1: